js/TabEnhancements/TabEnhancements.js

```diff
@@ -0,0 +1,66 @@
+var $ = require('jquery');
+
+/**
+ * Enhance tab.js behaviour
+*/
+class TabEnhancements {
+    /**
+     * Initialise
+     * @param {jQuery} $html - a jQuery wrapper of the html node
+     */
+    init ($html) {
+        this.$html = $html;
+
+        if(typeof $.fn.tab === 'undefined') {
+            console.warn('PULSAR: tab.js must be loaded in order to use TabEnhacements.js');
+        }
+
+        // Make sure tab panes are at least as high as the tab list (cms legacy tabs list)
+        this.$html.find('.tabs > .tabs__content > .tab__pane').css('min-height', this.$html.find('.tabs__list').height());
+
+        // Switch the first active tab to use <main> with the skip target
+        this.changeActiveTabElementToMain(this.$html.find('.tab__pane.is-active'));
+
+
+        // Make the current tabs .tab__content <main> and the previous tabs .tab__content <div>
+        // to avoid multiple <main>'s in the DOM at one time
+        this.$html.find('.nav-inline [data-toggle="tab"]').on('show.bs.tab', (event) => {
+            const $activeTab = this.$html.find($(event.target).attr('href'));
```

Comment:
Is there any benefit to just using `event.target` and `event.relatedTarget` instead of fetching the href here? Smells like a lot of the following methods might be able to work just fine this way?

Reply:
Not sure I follow, `event.target` and `event.relatedTarget` refer to the `a.nav-inline__link` of the new tab and the `a.nav-inline__link` of the old tab, not the tab itself. So we need to grab the href to find the `id` of the new/previous `div.tab__pane`.

Reply:
Ah ok, I thought it referred to the tab pane

Reply:
👍